src/sidebar/services/annotations.js

```diff
@@ -59,7 +61,66 @@ export default function annotations(
     searchClient.get({ uri: uris, group: groupId });
   }
 
+  /**
+   * Apply changes for the given `annotation` from its draft in the store (if
+   * any) and return a new object with those changes integrated.
+   */
+  function applyDraftChanges(annotation) {
+    const changes = {};
+    const draft = store.getDraft(annotation);
+
+    if (draft) {
+      changes.tags = draft.tags;
+      changes.text = draft.text;
+      changes.permissions = draft.isPrivate
+        ? privatePermissions(annotation.user)
+        : sharedPermissions(annotation.user, annotation.group);
+    }
+
+    // Integrate changes from draft into object to be persisted
+    return Object.assign({}, annotation, changes);
+  }
+
+  /**
+   * Save new (or update existing) annotation. On success,
+   * the annotation's `Draft` will be removed and the annotation added
+   * to the store.
+   */
+  function save(annotation) {
+    let saved;
+
+    const annotationWithChanges = applyDraftChanges(annotation);
+
+    if (isNew(annotation)) {
+      saved = api.annotation.create({}, annotationWithChanges);
+    } else {
+      saved = api.annotation.update(
+        { id: annotation.id },
+        annotationWithChanges
+      );
+    }
+
+    return saved.then(savedAnnotation => {
+      // Copy across internal properties which are not part of the annotation
+      // model saved on the server
+      savedAnnotation.$tag = annotation.$tag;
```

Comment:
I wonder why `$tag` was treated separately here. Provided that property is enumerable (which I think it should be) the loop below should catch it.

Reply:
Oh, good point. Blind copypasta. Will change.